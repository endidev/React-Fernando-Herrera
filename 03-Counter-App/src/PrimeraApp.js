// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types'; 

//fc 
const PrimeraApp = ( {saludo, subtitulo = 'Soy un subtitulo'} ) => {

    // const saludo = {
    //     nombre: 'Adrian',
    //     edad: 32
    // }
    
   

    return (
        <>
            <h1> { saludo } </h1>
            {/* <pre> { JSON.stringify(saludo, null, 3) } </pre> */}
            <p> {subtitulo} </p>
        </>
    )
}

// lo declaramos con mayuscula, pero aca la usamos 
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;