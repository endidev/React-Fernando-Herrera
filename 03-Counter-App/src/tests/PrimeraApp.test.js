import React from 'react';
import { getByText, render } from '@testing-library/react'
import '@testing-library/jest-dom'
import PrimeraApp from '../PrimeraApp';
import {shallow} from 'enzyme'

describe('Pruebas en PrimeraApp', () => {
    // test('debe mostrar un mensaje', () => {
    //     const saludo = 'Hola, soy endi';
    //     const { getByText } = render(<PrimeraApp saludo= {saludo} />)
    //     expect(getByText( saludo )).toBeInTheDocument;    
    // })

    test('debe mostrar <PrimeraAPP /> correctamente', () => {
        const saludo = 'Hola, soy endi';
        const wrapper = shallow(<PrimeraApp saludo={saludo}/>)

        expect( wrapper ).toMatchSnapshot();
    })
    
    test('debe mostrar subtitulo enviado por props', () => {
        const saludo = 'Hola, soy endi';
        const subTitulo = 'Soy un subtitulo';

        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo}
                subtitulo={subTitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();

        expect(textoParrafo).toBe(subTitulo);
    })
    
    
})