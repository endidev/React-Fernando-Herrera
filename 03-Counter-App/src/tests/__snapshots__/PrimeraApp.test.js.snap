// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en PrimeraApp debe mostrar <PrimeraAPP /> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy endi
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
